client/src/components/signup.jsx

```diff
@@ -32,8 +32,8 @@ class SignUp extends Component {
 
   handleSubmit = async () => {
     try {
-      const { username, password, email } = this.state;
-      await AmplifyAuth.signUp({ username, password, attributes: { email } });
+      const { name, username, password, email } = this.state;
+      await AmplifyAuth.signUp({ name, username, password, attributes: { email } });
```

Comment:
I think username and password are the only “core” attributes for Amplify/Cognito sign up. `name` should be inside the `attributes` object, like `email`.